Basilisk II fails to build with TUN/TAP support on CentOS 7

With 1c9b21f on CentOS 7, `configure` reports:

    checking whether TUN/TAP is supported... no

`config.log` shows:

    configure:15995: checking whether TUN/TAP is supported
    configure:16025: gcc -c -g -O2  conftest.c >&5
    In file included from conftest.c:94:0:
    /usr/include/linux/if.h:211:19: error: field 'ifru_addr' has incomplete type
       struct sockaddr ifru_addr;
                       ^
    /usr/include/linux/if.h:212:19: error: field 'ifru_dstaddr' has incomplete type
       struct sockaddr ifru_dstaddr;
                       ^
    /usr/include/linux/if.h:213:19: error: field 'ifru_broadaddr' has incomplete type
       struct sockaddr ifru_broadaddr;
                       ^
    /usr/include/linux/if.h:214:19: error: field 'ifru_netmask' has incomplete type
       struct sockaddr ifru_netmask;
                       ^
    /usr/include/linux/if.h:215:20: error: field 'ifru_hwaddr' has incomplete type
       struct  sockaddr ifru_hwaddr;

It appears that `sys/socket.h` must be included for `struct sockaddr`.

--- a/BasiliskII/src/Unix/configure.ac
+++ b/BasiliskII/src/Unix/configure.ac
@@ -906,6 +906,9 @@ dnl Check that the host supports TUN/TAP devices
 AC_CACHE_CHECK([whether TUN/TAP is supported],
   ac_cv_tun_tap_support, [
   AC_TRY_COMPILE([
+    #ifdef HAVE_SYS_SOCKET_H
+    #include <sys/socket.h>
+    #endif
     #if defined(HAVE_LINUX_IF_H) && defined(HAVE_LINUX_IF_TUN_H)
     #include <linux/if.h>
     #include <linux/if_tun.h>

